File: test_random.cpp
/*--***************************************************************************

    ...

***************************************************************************--*/

#include <algorithm>
#include <iostream>
#include "sorting.h"

void print_list(int* aList, int lenOfList);

int main(){

    const int len_list = 10;
    int rand_list[len_list];
    int std_sorted_list[len_list];
    bool is_identical = true;

    std::cout << "A random list:" << "\n";
    for (int i=0; i<len_list; i++){
       rand_list[i] = rand() % 20;
    }
    print_list(rand_list, len_list);
    std::cout << "\n";

    std::cout << "Standard results:" << "\n";
    std::copy(rand_list, rand_list + len_list, std_sorted_list);
    std::sort(std_sorted_list, std_sorted_list + len_list);
    print_list(std_sorted_list, len_list);
    std::cout << "\n";

    std::cout << "Sorted by my implementation:" << "\n";    
    quick_sort(rand_list, 0, len_list - 1);
    print_list(rand_list, len_list);
    std::cout << "\n";    

    std::cout << "My result is ";
    for (int i=0; i<len_list; i++){
        if (std_sorted_list[i] != rand_list[i]){
            is_identical = false;
            break;
        }
    }
    std::cout << std::boolalpha << is_identical << "\n";

    return 0;
}

void print_list(int* aList, int lenOfList){

    const int last = lenOfList - 1;

    std::cout << "[";
    for (int i=0; i<last; i++){
        std::cout << aList[i] << ", ";
    }
    std::cout << aList[last] << "]\n";
}

File: test_random.py
#!/usr/bin/env python


from __future__ import print_function
import random
from sorting import heap_sort as mysort


n = 10
rand_int_list = [random.randint(0, 20) for _ in range(n)]
print("A random list:")
print(rand_int_list, '\n')

print("StandarD results:")
print(sorted(rand_int_list), '\n')

mysort(rand_int_list)
print("Sorted by my implementation:")
print(rand_int_list, '\n')

is_identical = True if sorted(rand_int_list) == rand_int_list else False
print("My result is ", end='')
print(is_identical)
